tests/components/tplink_omada/test_switch.py

```diff
@@ -0,0 +1,88 @@
+"""Tests for TP-Link Omada switch entities."""
+from unittest.mock import MagicMock
+
+from tplink_omada_client.devices import OmadaSwitchPortDetails
+from tplink_omada_client.omadasiteclient import SwitchPortOverrides
+
+from homeassistant.components import switch
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+async def test_poe_switches(
+    hass: HomeAssistant,
+    mock_omada_site_client: MagicMock,
+    init_integration: MockConfigEntry,
+) -> None:
+    """Test PoE switch."""
+    poe_switch_mac = "54-AF-97-00-00-01"
+    for i in range(1, 9):
+        await _test_poe_switch(
+            hass,
+            mock_omada_site_client,
+            f"switch.test_poe_switch_port_{i}_poe",
+            poe_switch_mac,
+            i,
+        )
+
+
+async def _test_poe_switch(
+    hass: HomeAssistant,
+    mock_omada_site_client: MagicMock,
+    entity_name: str,
+    network_switch_mac: str,
+    port: int,
+) -> None:
+    entity_registry = er.async_get(hass)
+
+    poe_enable: bool = True
+
+    async def assert_update_switch_port(
+        device, port, overrides: SwitchPortOverrides = None
+    ):
+        assert device
+        assert device.mac == network_switch_mac
+        assert port
+        assert port.port == port
+        assert overrides
+        assert overrides.enable_poe == poe_enable
+
+        return OmadaSwitchPortDetails({"poe": 1 if overrides.enable_poe else 0})
+
+    entity = hass.states.get(entity_name)
+    assert entity
+    assert entity.state == "on"
```

Comment:
A more complete test of the state + attributes may be a good idea, you can add snapshots to avoid polluting the Python code with hard coded dictionaries: https://developers.home-assistant.io/blog/2023/02/20/snapshot-testing

Reply:
snapshots may be one of the coolest things I've seen in a while.